doc/function_types.md

```diff
@@ -508,6 +508,7 @@ The following optimizers can be used with differentiable functions:
  - [AdaDelta](#adadelta)
  - [AdaGrad](#adagrad)
  - [Adam](#adam)
+ - [QHAdam](#qhadam)
```

Comment:
Just a minor note, maybe we can address it during emrge---all the other optimizers are arranged alphabetically, we should probably keep it like that.